### PR TITLE
Automate checkstyle

### PR DESCRIPTION
**Pull request by J0c4**
Add files needed to run checkstyle using gradle and include this as part of a GitHub workflow

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -16,4 +17,22 @@ test {
         events "failed"
         exceptionFormat "full"
     }
-}
```
Comment (jpgonzales27) [inline build.gradle:19]: this key is excess and generating the error

File: build.gradle [hunk 2]
```diff
@@ -16,4 +17,22 @@ test {
...
+
+checkstyle {
+    toolVersion '8.34'
+    configFile file('config/checkstyle/checkstyle.xml')
+    checkstyleMain.source = "src/main/"
+    checkstyleMain.excludes = ["**/src/test/**"]
+}
+
+checkstyleMain {
+    source ='src/main/java'
```
Comment (cristianfchq) [inline build.gradle:30]: Is missing space between "=" and 'src/main/java'

File: build.gradle [hunk 3]
```diff
@@ -16,4 +17,22 @@ test {
...
+}
+
+checkstyleTest {
+    source ='src/main/java'
```
Comment (cristianfchq) [inline build.gradle:34]: Is missing space between **"="** and **'src/main/java'**

File: .github/workflows/gradle.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up JDK 1.8
+      uses: actions/setup-java@v1
+      with:
+        java-version: 1.8
```
Comment (edsonarios) [inline .github/workflows/gradle.yml:20]: Will it work with higher versions of java? like java 11 or 15?

File: build.gradle (hunk 1 above)
Comment (JRojas999) [inline build.gradle:19]: A blank line is missing.

Comment (J0c4) [reply build.gradle:19]: Talked with Joel, there is no blank line missing. No need for changes

**Review by edsonarios — COMMENTED** [review]
Will it work with higher versions of java? like java 11 or 15?

File: .github/workflows/gradle.yml (hunk 4 above)
Comment (J0c4) [reply .github/workflows/gradle.yml:20]: If we avoid using specific feature from java 11 or 15, we should be ok. However, we can consider to upgrade the version we are using in the future

**Review by Raymundogs — APPROVED** [review]

**Review by jpgonzales27 — APPROVED** [review]

**Pull request merged**